### PR TITLE
[Event Hubs] Function Test Fix

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to fix an assumption in the Functions extension package that the same Storage suffix is valid in all clouds.  The implementation supports the Azure public cloud only, while the test was asserting that the scenario was valid for all clouds.  Going forward, the test will be ignored when the current execution environment is not a supported cloud.

Also riding along is a tweak to a processor test to extract additional error details for failure investigation.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs [hunk 1]
```diff
@@ -252,6 +254,17 @@ public async Task CanSendAndReceive_BlobServiceUri_InConfiguration()
         [Test]
         public async Task CanSendAndReceive_AccountName_InConfiguration()
         {
+            // Use of an account name assumes an endpoint suffix that is only valid in some Azure
+            // cloud environments. If the current execution environment uses a different suffix,
+            // ignore the test.
+
+            var defaultSuffix = StorageClientProvider<object, ClientOptions>.DefaultStorageEndpointSuffix;
+
+            if (string.Equals(EventHubsTestEnvironment.Instance.StorageEndpointSuffix, defaultSuffix, StringComparison.OrdinalIgnoreCase))
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:263]:
```suggestion
            if (!string.Equals(EventHubsTestEnvironment.Instance.StorageEndpointSuffix, defaultSuffix, StringComparison.OrdinalIgnoreCase))
```

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs]: uh... yeah, that's an embarrassing one.

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs [hunk 2]
```diff
@@ -252,6 +254,17 @@ public async Task CanSendAndReceive_BlobServiceUri_InConfiguration()
...
         {
+            // Use of an account name assumes an endpoint suffix that is only valid in some Azure
+            // cloud environments. If the current execution environment uses a different suffix,
+            // ignore the test.
+
+            var defaultSuffix = StorageClientProvider<object, ClientOptions>.DefaultStorageEndpointSuffix;
+
+            if (!string.Equals(EventHubsTestEnvironment.Instance.StorageEndpointSuffix, defaultSuffix, StringComparison.OrdinalIgnoreCase))
+            {
+                Assert.Ignore($"This test can only be run in the Azure cloud associated  with the suffix: `{defaultSuffix}`.");
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:265]:
```suggestion
                Assert.Ignore($"This test can only be run in the Azure cloud associated with the suffix: `{defaultSuffix}`.");
```

**Comment by jsquire** [comment]
azp run net - eventhub - tests

**Review by christothes — APPROVED** [review]

**Pull request merged**